tools/mo/openvino/tools/mo/front/mxnet/arange_like_replacer.py

```diff
@@ -0,0 +1,148 @@
+from openvino.tools.mo.ops.elementwise import Add, Div, Mul
+from openvino.tools.mo.ops.gather import Gather
+from openvino.tools.mo.ops.range import Range
+from openvino.tools.mo.ops.reshape import Reshape
+from openvino.tools.mo.ops.shape import Shape
+from openvino.tools.mo.ops.slice import Slice
+from openvino.tools.mo.ops.squeeze import Squeeze
+from openvino.tools.mo.ops.tile import Tile
+
+
+class ArangeLikeReplacer(FrontReplacementOp):
+    op = 'arange_like'
+    enabled = True
+
+    def replace_sub_graph(self, graph: Graph, match: dict):
+        node = match['op']
+        name = node.soft_get('name', node.id)
+        axis = node.axis
+        data_out_port = node.in_port(0).get_source()
+        input_shape_node = Shape(graph, {'name': name + '/ShapeOf'}).create_node()
+        range_node = create_op_with_const_inputs(graph, Range, {0: mo_array(node.start),
+                                                                2: mo_array(node.step)}, {'name': name + '/Range'})
+        node.in_port(0).disconnect()
+        input_shape_node.in_port(0).connect(data_out_port)
+
+        if axis is not None:
+            '''
+            Replace arange_like op to subgraph:
+            Shape - Gather - Range
+            '''
+            gather_node = create_op_with_const_inputs(graph, Gather, {1: int64_array([axis]),
+                                                                      2: int64_array(0)},
+                                                      {'name': name + '/Gather'})
+            input_shape_node.out_port(0).connect(gather_node.in_port(0))
+            gather_node.out_port(0).connect(range_node.in_port(1))
+            node.out_port(0).get_connection().set_source(range_node.out_port(0))
+            rename_nodes([(node, name + '/ShouldBeDeleted'), (range_node, name)])
+        else:
+            r'''
+            Replace arange_like op to subgraph:
+                    |
+                    | -------------- |
+                    |                |
+                 Reshape             |
+                    |                 |
+                 ShapeOf           ShapeOf  
+                    |                |
+                  Range              |
+                    |                |
+                 Reshape ----------- | 
+                    |
+            '''
+
+            reshape_forward_node = create_op_with_const_inputs(graph, Reshape, {1: int64_array([-1])},
+                                                               {'name': name + '/Reshape_forward'})
+
+            reshape_backward_node = Reshape(graph, {'name': name + '/Reshape_backward'}).create_node()
+            shape_of_node = Shape(graph, {'name': reshape_forward_node.name + '/ShapeOf'}).create_node()
+
+            reshape_forward_node.in_port(0).connect(data_out_port)
+            reshape_forward_node.out_port(0).connect(shape_of_node.in_port(0))
+            shape_of_node.out_port(0).connect(range_node.in_port(1))
```

Comment:
nit: you insert below Squeeze for `in_port(0)` but for `in_port(1)` you don't do that. For me it's unclear why this works since `ShapeOf` according to spec always returnd 1D array not a scalar

Reply:
Discussed personally. `in_port(0)` of Range is always a scalar constant regardless of `node.start` value. Comment resolved.